### PR TITLE
Bug 2103975 - The ODF dashboard showing the to unpriviledged users

### PR DESCRIPTION
**Pull request by TimothyAsirJeyasing**
Signed-off-by: Timothy Asir Jeyasingh <tjeyasin@redhat.com>

### CODE REVIEW

**Comment by TimothyAsirJeyasing** [comment]
@bipuladh , @GowthamShanmugam , Please review

**Comment by TimothyAsirJeyasing** [comment]
@bipuladh , @GowthamShanmugam, @SanjalKatiyar Please review

File: packages/odf/components/odf-dashboard/dashboard.tsx [hunk 1]
```diff
@@ -63,6 +65,13 @@ const ODFDashboardPage: React.FC<ODFDashboardPageProps> = (props) => {
   const { t } = useCustomTranslation();
   const hasMCG = useFlag(MCG_FLAG);
   const title = t('Data Foundation');
+  const [canCreateODF] = useAccessReview({
+    group: ODFStorageSystem.apiGroup,
+    resource: ODFStorageSystem.plural,
+    verb: 'create',
+    namespace: null,
```
Comment (GowthamShanmugam) [inline packages/odf/components/odf-dashboard/dashboard.tsx:72]:
```suggestion
    namespace: CEPH_STORAGE_NAMESPACE,
```

File: packages/odf/components/odf-dashboard/dashboard.tsx [hunk 2]
```diff
@@ -118,17 +127,21 @@ const ODFDashboardPage: React.FC<ODFDashboardPageProps> = (props) => {
...
-        pages={pages}
-        resource={{
-          kind: ODFStorageSystemMock.kind,
-          apiVersion: `${ODFStorageSystemMock.apiGroup}/${ODFStorageSystemMock.apiVersion}`,
-        }}
-      />
+      {canCreateODF && (
+        <>
+          <Helmet>
+            <title>{title}Testing 12345</title>
```
Comment (GowthamShanmugam) [inline packages/odf/components/odf-dashboard/dashboard.tsx:133]:
```suggestion
            <title>{title}</title>
```

File: packages/odf/components/odf-dashboard/dashboard.tsx [hunk 3]
```diff
@@ -118,17 +127,21 @@ const ODFDashboardPage: React.FC<ODFDashboardPageProps> = (props) => {
...
-      </Helmet>
-      <PageHeading title={title} />
-      <HorizontalNav
-        pages={pages}
-        resource={{
-          kind: ODFStorageSystemMock.kind,
-          apiVersion: `${ODFStorageSystemMock.apiGroup}/${ODFStorageSystemMock.apiVersion}`,
-        }}
-      />
+      {canCreateODF && (
```
Comment (GowthamShanmugam) [inline packages/odf/components/odf-dashboard/dashboard.tsx:130]: I worried this will lead to a blank page

Comment (TimothyAsirJeyasing) [reply packages/odf/components/odf-dashboard/dashboard.tsx:130]: Could you point me the access denied url for this?

Comment (TimothyAsirJeyasing) [reply packages/odf/components/odf-dashboard/dashboard.tsx:130]: For other pages i have seen a "No <page> found"
I hope we can not display like "No ODF found". So shall we display some meaningful message

File: packages/odf/components/odf-dashboard/dashboard.tsx (hunk 2 above)
Comment (TimothyAsirJeyasing) [reply packages/odf/components/odf-dashboard/dashboard.tsx:133]: Will do the change

File: packages/odf/components/odf-dashboard/dashboard.tsx (hunk 1 above)
Comment (TimothyAsirJeyasing) [reply packages/odf/components/odf-dashboard/dashboard.tsx:72]: Will change this

File: packages/odf/components/odf-dashboard/dashboard.tsx (hunk 3 above)
Comment (SanjalKatiyar) [reply packages/odf/components/odf-dashboard/dashboard.tsx:130]: as I said before...
for error message use useAccessReview hook and https://github.com/red-hat-storage/odf-console/blob/8cd4a6dcd82037f4a2776a6c21cc1ce8e17c55ea/packages/shared/generic/status-box.tsx#L120 component. That is similar to PV page under Storage...

use loading state as well returned by useAccessReview as well to display loadingcomponent...

Comment (TimothyAsirJeyasing) [reply packages/odf/components/odf-dashboard/dashboard.tsx:130]: Sure, will do this

File: packages/odf/components/odf-dashboard/dashboard.tsx (hunk 2 above)
Comment (TimothyAsirJeyasing) [reply packages/odf/components/odf-dashboard/dashboard.tsx]: Done

**Comment by TimothyAsirJeyasing** [comment]
@GowthamShanmugam , @SanjalKatiyar Please review, I have updated as per your suggestions

**Comment by TimothyAsirJeyasing** [comment]
image

**Comment by bipuladh** [comment]
> image

Can we make this similar to how OCP looks?

**Comment by TimothyAsirJeyasing** [comment]
image
image

@bipuladh Please review

**Pull request merged**